### PR TITLE
Truncated element

### PR DESCRIPTION
**Pull request by AndreiBaicu26**
To succesfully implement the [Breadcrumbs](https://github.com/adobe/spectrum-web-components/discussions/4100) component we would require the truncated element from UEC. We copied the component in SWC so we do not create a new dependency to UEC. 

## Related issue(s)

https://github.com/adobe/spectrum-web-components/discussions/4100

## How has this been tested?

-   [x] _Test case 1_
    1. Opened storybook 
    2. Hover over truncated elements
    3. The tooltip appears.
-   [x] _Test case 2_
    1.  Opened storybook 
    2. Hover over non-truncated elements
    3. The tooltip should not appear

## Screenshots (if appropriate)

image

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

File: tools/truncated/package.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
+{
+    "name": "@spectrum-web-components/truncated",
+    "version": "0.41.1",
```
Comment (Westbrook) [inline tools/truncated/package.json:3]: While technically correct, the performance test jobs for Chrome and Firefox will continue to fail why this is not `0.0.1`.

File: tools/truncated/package.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    "devDependencies": {
+        "@open-wc/testing": "^3.2.0",
+        "@web/test-runner-commands": "^0.9.0"
+    },
+    "types": "./src/index.d.ts",
+    "customElements": "custom-elements.json",
+    "sideEffects": [
+        "./sp-*.js",
+        "./**/*.dev.js",
```
Comment (Westbrook) [inline tools/truncated/package.json:73]: This is wrong like this somewhere in our tooling 🙈. Please you remove the duplicate.

File: tools/truncated/package.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    "dependencies": {
+        "@spectrum-web-components/base": "^0.41.0",
+        "@spectrum-web-components/overlay": "^0.41.0",
+        "@spectrum-web-components/styles": "^0.41.0",
+        "@spectrum-web-components/tooltip": "^0.41.0"
+    },
+    "devDependencies": {
+        "@open-wc/testing": "^3.2.0",
+        "@web/test-runner-commands": "^0.9.0"
+    },
```
Comment (Westbrook) [inline tools/truncated/package.json:68]: These are not needed in this context.

File: tools/truncated/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+When looking to leverage the `Truncated` base class as a type and/or for extension purposes, do so via:
+
+```
+import { Truncated } from '@spectrum-web-components/truncated';
+```
+
+
+## Example
+
+```
```
Comment (Westbrook) [inline tools/truncated/README.md:33]:
```suggestion
```html
```
Should activate the live demo in the docs site.

File: tools/truncated/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+<sp-truncated>This will truncate into a tooltip if there isn't enough space for it.</sp-truncated>
+```
+
+### With specific overflow content
+
+By default, tooltip text will be extracted from overflowing content. To provide your own overflow content, slot it into "overflow":
+
+```
```
Comment (Westbrook) [inline tools/truncated/README.md:41]:
```suggestion
```html
```
Should activate the live demo in the docs site.

**Review by Westbrook — COMMENTED** [review]
There are a couple of interactions that require testing before this code meets the higher requirements of this library, see: https://output.circle-artifacts.com/output/job/90cd7104-b24e-4b7d-ae41-865a34e8dceb/artifacts/0/coverage/lcov-report/tools/truncated/src/Truncated.ts.html

File: tools/truncated/package.json (hunk 1 above)
Comment (AndreiBaicu26) [reply tools/truncated/package.json:3]: Will the CI job upgrade it automatically to latest version ? I manually changed it from 0.0.1 to 0.41.1 so it matches the other components 😅

Comment (Westbrook) [reply tools/truncated/package.json:3]: Release time tooling ensures that things match when this is merged and release is ready to go.

Comment (AndreiBaicu26) [reply tools/truncated/package.json:3]: Alrighty, will leave it to 0.0.1

**Comment by Westbrook** [comment]
Can you catch this up to `main` for the review?

**Review by Westbrook — APPROVED** [review]
LGTM!

**Pull request merged**